src/actions/swapParent.ts

```diff
@@ -20,28 +20,48 @@ const swapParent = (state: State) => {
   // If the cursor is at the root, do nothing.
   if (!parent.length) return state
 
-  const childValue = getThoughtById(state, head(cursor))?.value
-  const parentValue = getThoughtById(state, head(parent))?.value
-  if (!childValue || !parentValue) return state
-
-  const reducers = [
-    editThought({
-      oldValue: childValue,
-      newValue: parentValue,
-      path: simplifyPath(state, cursor),
-      force: true,
-    }),
-    editThought({
-      oldValue: parentValue,
-      newValue: childValue,
-      path: simplifyPath(state, parent),
-    }),
-  ]
-
-  return reducerFlow(reducers)(state)
+  const childId = head(cursor)
+  const parentId = head(parent)
+
+  const childThought = getThoughtById(state, childId)
+  const parentThought = getThoughtById(state, parentId)
+  if (!childThought || !parentThought) return state
+
+  // Get children of the child thought that will be moved
+  const childChildren = getChildrenRanked(state, childId)
+
+  // First move the child to replace its parent's position
+  let stateNew = moveThought(state, {
```

Comment:
(Feel free to ignore if this was intentionally temporary!)

I recommend using utilities such as `reducerFlow` to compose reducers. Mutation is considered a bad practice, and this project adheres to functional programming principles.